corporate/lib/stripe.py

```diff
@@ -1096,25 +1121,58 @@ def process_initial_upgrade(
                             _("Please add a credit card before starting your free trial."),
                         )
 
+            event_time = billing_cycle_anchor
+            if should_schedule_upgrade_for_legacy_remote_server:  # nocoverage
+                assert remote_server_legacy_plan is not None
+                if charge_automatically:
+                    remote_server_legacy_plan.charge_automatically = True
+                    # Ensure customers not paying via invoice have a default payment method set.
+                    stripe_customer = stripe_get_customer(customer.stripe_customer_id)
+                    if not stripe_customer_has_credit_card_as_default_payment_method(
+                        stripe_customer
+                    ):
+                        raise BillingError(
+                            "no payment method",
+                            _("Please add a credit card to schedule upgrade."),
+                        )
+
+                # Settings status > CustomerPLan.LIVE_STATUS_THRESHOLD makes sure we don't have
+                # to worry about this plan being used for any other purpose.
+                # NOTE: This is the 2nd plan for the customer.
+                plan_params["status"] = CustomerPlan.NEVER_STARTED
+                event_time = timezone_now().replace(microsecond=0)
+
+                # Switch to the new plan at plan end date.
+                assert remote_server_legacy_plan.end_date == billing_cycle_anchor
+                remote_server_legacy_plan.status = CustomerPlan.SWITCH_PLAN_TIER_AT_PLAN_END
+                remote_server_legacy_plan.price_per_license = price_per_license
```

Comment:
Why do we change the price on the legacy plan here?

Reply:
To show it correctly on the billing page for customers before the new plan starts. `next_invoice_date` is `None` here to avoid any payments.

Reply:
OK this is a bit of a hack, right? Structurally, we probably need some other refactor to send the price for the upcoming plan in this case rather than creating these somewhat evil non-invoiceable license ledge entries.